### PR TITLE
Use postgres advisory lock to ensure that only one server does the rotations

### PR DESCRIPTION
**Pull request by ismarc**
#### What does this PR do?
Enable advisory locks around secret rotation to prevent conflicts and race conditions on multiple servers trying to rotate secrets.
#### Any background context you want to provide?
When Conjur does secret rotations, it fetches all secrets that need rotation from the database, then processes the list one by one, rotating each then updating the DB.  If multiple Conjur instances point to the same DB, this allows for race conditions and multiple updates to happen on each rotation, even if each secret is locked individually.  This provides a lock around the rotation process itself so only one server will do the rotation each time.  In the future, the process of determining which secret to rotate, and fetching them from the DB one at a time could remove the need for the advisory lock.
#### How should this be manually tested?
Provide the same CONJUR_DATA_KEY and postgres URL to two different Conjur instances, configure a rotation and observe the secret only being rotated by one server on each increment
#### Has the Version and Changelog been updated?
Covered by existing tests
> Can we make a blog post, video, or animated GIF of this?
Maybe?  Wouldn't be very exciting
> Has this change been documented (Readme, docs, etc.)?
This particular change doesn't have any needed associated documentation.  What it enables (multiple Conjur instances to a single Postgres db) should be documented once fully verified.
> Does the knowledge base need an update?
No

### CODE REVIEW

File: app/domain/rotation/master_rotator.rb [hunk 1]
```diff
@@ -28,7 +31,17 @@ def rotate_every(seconds)
     end
 
     def rotate_all
-      scheduled_rotations.each(&:run)
+      # Attempt an advisory lock to prevent other servers from trying to rotate
+      # at the same time or immediately after another server rotated the secret.
+      # The same `id` needs to be used across all servers talking to the same
+      # database, but can be any unused integer.
+      id = 8675309
```
Comment (sgnn7) [inline app/domain/rotation/master_rotator.rb]: This needs to be something static and definitely not within the rotate function itself. Can we take a hash of something stable to generate it?

File: app/domain/rotation/master_rotator.rb [hunk 2]
```diff
@@ -28,7 +31,17 @@ def rotate_every(seconds)
...
+      @db.try_advisory_lock(id) do
+        scheduled_rotations.each(&:run)
+      end
+    rescue => e
+      set_retry_expiration
+      log_error(e)
```
Comment (sgnn7) [inline app/domain/rotation/master_rotator.rb]: This loop seems a tad bit odd - we still want to propagate an error up if the rotation itself fails

Comment (sgnn7) [reply app/domain/rotation/master_rotator.rb]: Also, this really should be a warning if it was just the locking failure, not an error probably

File: app/domain/rotation/master_rotator.rb (hunk 1 above)
Comment (ismarc) [reply app/domain/rotation/master_rotator.rb]: Updated to use a string specified under initializers for the rotator that gets converted to integer at point in time (used the string "possum" for now).

File: app/domain/rotation/master_rotator.rb (hunk 2 above)
Comment (ismarc) [reply app/domain/rotation/master_rotator.rb]: After reviewing the history, this was a bad copy/paste from deeper in the stack where the error handling was needed, it's been pulled out.

File: config/initializers/rotator.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Rails.application.configure do
+  config.rotator_lock_name = "conjur"
+end
+
```
Comment (sgnn7) [inline config/initializers/rotator.rb]: @jtuttle / @ismarc Drive-by comment: Stray newline
